core/state/state_object.go

```diff
@@ -167,6 +167,13 @@ func (s *stateObject) GetState(key common.Hash) common.Hash {
 	return s.GetCommittedState(key)
 }
 
+func (s *stateObject) GetOriginState(key common.Hash) common.Hash {
+	if value, cached := s.originStorage[key]; cached {
+		return value
+	}
+	return common.Hash{}
+}
```

Comment:
Could be somehow better signal that `originStorage` is a cache and not a fully reliable source of truth? Maybe renaming to `originStorageCache` and also rename the method to `GetOriginStateCache` or something? 

I don't like the rename to `GetOriginStateCache` but L309 of `core/state/statedb.go` was quite surprising to me, and then I realized it's because it's only asking for a cache.

And even a more important reason to be super-clear about this `GetOriginState` being a cache, is that this method is public. If I see that name in an API and returns `common.Hash{}` I'd assume it doesn't exist, not that "potentially I don't know".